pkg/controller/dataimportcron-controller.go

```diff
@@ -0,0 +1,657 @@
+		Name:  "cdi-source-update-poller",
+		Image: r.image,
+		Command: []string{
+			"/usr/bin/cdi-source-update-poller",
+			"-ns", cron.Namespace,
+			"-cron", cron.Name,
+			"-url", *regSource.URL,
+		},
+		ImagePullPolicy: corev1.PullPolicy(r.pullPolicy),
+	}
+
+	hasCertConfigMap := regSource.CertConfigMap != nil && *regSource.CertConfigMap != ""
+	if hasCertConfigMap {
+		vm := corev1.VolumeMount{
+			Name:      CertVolName,
+			MountPath: common.ImporterCertDir,
+		}
+		container.VolumeMounts = []corev1.VolumeMount{vm}
+		container.Command = append(container.Command, "-certdir", common.ImporterCertDir)
+	}
+
+	if regSource.SecretRef != nil && *regSource.SecretRef != "" {
+		container.Env = append(container.Env,
+			corev1.EnvVar{
+				Name: common.ImporterAccessKeyID,
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: *regSource.SecretRef,
+						},
+						Key: common.KeyAccess,
+					},
+				},
+			},
+			corev1.EnvVar{
+				Name: common.ImporterSecretKey,
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: *regSource.SecretRef,
+						},
+						Key: common.KeySecret,
+					},
+				},
+			},
+		)
+	}
+
+	if insecureTLS {
+		container.Env = append(container.Env,
+			corev1.EnvVar{
+				Name:  common.InsecureTLSVar,
+				Value: "true",
+			},
+		)
+	}
+
+	var successfulJobsHistoryLimit int32 = 0
+	var failedJobsHistoryLimit int32 = 0
+
+	job := &v1beta1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GetCronJobName(cron),
+			Namespace: r.cdiNamespace,
+		},
+		Spec: v1beta1.CronJobSpec{
+			Schedule:                   cron.Spec.Schedule,
+			ConcurrencyPolicy:          v1beta1.ForbidConcurrent,
+			SuccessfulJobsHistoryLimit: &successfulJobsHistoryLimit,
+			FailedJobsHistoryLimit:     &failedJobsHistoryLimit,
+			JobTemplate: v1beta1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							RestartPolicy:      corev1.RestartPolicyNever,
+							Containers:         []corev1.Container{container},
+							ServiceAccountName: "cdi-cronjob",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if hasCertConfigMap {
+		vol := corev1.Volume{
+			Name: CertVolName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: *regSource.CertConfigMap,
+					},
+				},
+			},
+		}
+		job.Spec.JobTemplate.Spec.Template.Spec.Volumes = []corev1.Volume{vol}
+	}
+
+	return job, nil
+}
+
+func newSourceDataVolume(cron *cdiv1.DataImportCron, dataVolumeName string) *cdiv1.DataVolume {
+	var digestedURL string
+	dv := cron.Spec.Template.DeepCopy()
+	if isURLSource(cron) {
+		digestedURL = *dv.Spec.Source.Registry.URL + "@" + cron.Annotations[AnnSourceDesiredDigest]
```

Comment:
I forget does the URL contain the tag element as well? And if that even matter when you specify the sha. If the functional tests pass we would know I guess.

Reply:
Actually just saw you take the tag off. so we are good.